corva/application.py

```diff
@@ -0,0 +1,52 @@
+from typing import Callable, List, Optional
+
+from corva.middleware.unpack_context import unpack_context
+from corva.models.base import BaseContext
+from corva.types import MIDDLEWARE_CALL_TYPE, MIDDLEWARE_TYPE
+
+
+def wrap_call_in_middleware(
+     call: Callable,
+     middleware: Optional[List[MIDDLEWARE_TYPE]] = None
+) -> MIDDLEWARE_CALL_TYPE:
+    def wrapper_factory(mw, call):
+        def wrapper(ctx):
+            return mw(ctx, call)
+
+        return wrapper
+
+    middleware = middleware or []
+
+    for mw in reversed(middleware):
+        call = wrapper_factory(mw, call)
+
+    return call
+
+
+class Corva:
+    def __init__(
+         self,
+         *,
+         middleware: Optional[List[MIDDLEWARE_TYPE[BaseContext]]] = None
+    ):
+        self.user_middleware = middleware or []
+
+    def get_middleware_stack(
+         self,
+         middleware: Optional[List[MIDDLEWARE_TYPE[BaseContext]]] = None
```

Comment:
I don't think this should be Optional/None. It can default to an empty list.

Reply:
it is not good to use mutable types as default arguments

we can do the following as alternative:

```suggestion
         middleware: typing.Sequence[MIDDLEWARE_TYPE[BaseContext]] = ()
```

Reply:
previous is better. Let's work with Optional, rather than with sequences